src/services/pecServerClientFactory.ts

```diff
@@ -0,0 +1,45 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as O from "fp-ts/lib/Option";
+import * as TE from "fp-ts/lib/TaskEither";
+import { PecBearerGeneratorT } from "src/types/token";
+import { IPecServerClient, pecServerClient } from "../clients/pecserver";
+import {
+  PecServersConfig,
+  PecServerConfig,
+  getHttpsApiFetchWithBearer
+} from "../config";
+import { IPecServerClientFactoryInterface } from "./IPecServerClientFactory";
+
+const findById = (sources: PecServersConfig, serviceId: NonEmptyString) =>
+  O.fromNullable(
+    Object.values(sources)
+      .filter(PecServerConfig.is)
+      .find(c => c.serviceId === serviceId)
+  );
+
+export default class PecServerClientFactory
+  implements IPecServerClientFactoryInterface {
+  constructor(private readonly pecConfigs: PecServersConfig) {}
+
+  /**
+   * This method returns a specific pec server client based on
+   * provider serviceId
+   * {@inheritDoc}
+   */
+  public getClient(
+    bearerGenerator: PecBearerGeneratorT,
+    maybeServiceId: O.Option<NonEmptyString>
```

Comment:
Code design:

it's preferrable not to pass monadic types as parameters, as it's usually an indicator for "this function hides some business decision". That's the case: we're using `poste` as default PEC operator, but such info isn't explicit for the caller. 

What I would do instead:
* expect a whole _pecConfig_ struct as parameter
* explicitly declare `this.pecConfigs.poste` as default for such parameter
* let the caller to resolve proper _pecConfig_ from the given _serviceId_

Makes sense?

Reply:
Otherwise, if we think it's this function's responsibility to resolve _pecConfig_ from _serviceId_, we can:
* define _serviceId_ as `serviceId?: NonEmptyString`
* use `fromNullable` inside the body, to obtain a monadic type
* store `this.pecConfigs.poste` in a `DEFAULT_PEC_CONFIG` class constant, so that it would be explicit